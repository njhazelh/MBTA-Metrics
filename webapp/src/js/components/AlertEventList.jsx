import React from 'react';
import { connect } from 'react-redux';
import { Table } from 'reactstrap';

import AlertEvent from './AlertEvent';


class AlertEventList extends React.Component {
  render() {
    const { alertEvents } = this.props;
    const table = alertEvents.length === 0
      ? (<tr style={{ textAlign: 'center' }}>
        <td colSpan="17" className="py-5">No Data Loaded</td>
      </tr>)
      : alertEvents.map(alertEvent =>
        <AlertEvent data={alertEvent} key={alertEvent.id} />,
        );

    return (
      <Table responsive striped bordered className="table-sm mt-2">
        <thead className="thead-default">
          <tr>
            <th>Trip ID</th>
            <th>Date</th>
            <th>Time</th>
            <th>Day</th>
            <th>Route</th>
            <th>Stop</th>
            <th>Direction</th>
            <th>Short Name</th>
            <th>Scheduled Dep.</th>
            <th>Actual Dep.</th>
            <th>Delay</th>
            <th>Alert?</th>
            <th>Deserves Alert?</th>
            <th>Alert Delay</th>
            <th>Alert Text</th>
            <th>Predicted Delay</th>
            <th>Delay Accuracy</th>
          </tr>
        </thead>
        <tbody>
          {table}
        </tbody>
      </Table>
    );
  }
}

export default connect(data => ({ alertEvents: data.alertFilteredData }))(AlertEventList);
