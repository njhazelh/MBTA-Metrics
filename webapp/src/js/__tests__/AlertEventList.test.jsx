/* eslint-env jest */

import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Moment from 'moment';

import AlertEventList from '../components/AlertEventList';
import store from '../store';
import { getAlertEventsSuccess } from '../actions/alertEvent-actions';
import * as filterActions from '../actions/filterActions';

test('AlertEventList should render data', () => {
  const component = renderer.create(
    <Provider store={store}><AlertEventList /></Provider>,
  );
  store.dispatch(filterActions.setAllDirectionFilters(true));
  store.dispatch(filterActions.setAllLineFilters(true));
  store.dispatch(filterActions.setAllAlertFilters(true));

  store.dispatch(getAlertEventsSuccess([
    { id: 1, direction: 'inbound' },
    { id: 2, direction: 'outbound', deserves_alert: true, alert_issued: true },
    { id: 3, direction: 'inbound', delay: -14.999, actual_departure: Moment('2017-04-26') },
    { id: 4, alert_delay: 5.1111, scheduled_departure: Moment('2017-04-26') },
    { id: 5, date: Moment('2017-04-26'), time: Moment('2017-04-26') },
  ]));
  expect(component.toJSON()).toMatchSnapshot();

  store.dispatch(filterActions.setDirectionFilter('inbound', false));
  store.dispatch(filterActions.setLineFilter('CR-Middleborough', false));
  store.dispatch(filterActions.setAlertFilter('accurate', false));
  expect(component.toJSON()).toMatchSnapshot();

  store.dispatch(filterActions.setAllDirectionFilters(false));
  store.dispatch(filterActions.setAllLineFilters(false));
  store.dispatch(filterActions.setAllAlertFilters(false));
  expect(component.toJSON()).toMatchSnapshot();
});
