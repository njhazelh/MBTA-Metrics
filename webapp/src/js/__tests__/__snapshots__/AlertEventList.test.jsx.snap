// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertEventList should render data 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table-sm mt-5 table table-bordered table-striped"
  >
    <thead
      className="thead-default"
    >
      <tr>
        <th>
          Trip ID
        </th>
        <th>
          Date
        </th>
        <th>
          Time
        </th>
        <th>
          Day
        </th>
        <th>
          Route
        </th>
        <th>
          Stop
        </th>
        <th>
          Direction
        </th>
        <th>
          Short Name
        </th>
        <th>
          Scheduled Dep.
        </th>
        <th>
          Actual Dep.
        </th>
        <th>
          Delay
        </th>
        <th>
          Alert?
        </th>
        <th>
          Deserves Alert?
        </th>
        <th>
          Alert Delay
        </th>
        <th>
          Alert Text
        </th>
        <th>
          Predicted Delay
        </th>
        <th>
          Delay Accuracy
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td>
          inbound
        </td>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          No
        </td>
        <td>
          No
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td>
          outbound
        </td>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          Yes
        </td>
        <td>
          Yes
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td>
          inbound
        </td>
        <td />
        <td>
          
        </td>
        <td>
          12:00 AM
        </td>
        <td>
          -15 min
        </td>
        <td>
          No
        </td>
        <td>
          No
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td />
        <td />
        <td>
          12:00 AM
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          No
        </td>
        <td>
          No
        </td>
        <td>
          5 min
        </td>
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td>
          04/26/2017
        </td>
        <td>
          12:00 AM
        </td>
        <td>
          Wednesday
        </td>
        <td />
        <td />
        <td />
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          No
        </td>
        <td>
          No
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`AlertEventList should render data 2`] = `
<div
  className="table-responsive"
>
  <table
    className="table-sm mt-5 table table-bordered table-striped"
  >
    <thead
      className="thead-default"
    >
      <tr>
        <th>
          Trip ID
        </th>
        <th>
          Date
        </th>
        <th>
          Time
        </th>
        <th>
          Day
        </th>
        <th>
          Route
        </th>
        <th>
          Stop
        </th>
        <th>
          Direction
        </th>
        <th>
          Short Name
        </th>
        <th>
          Scheduled Dep.
        </th>
        <th>
          Actual Dep.
        </th>
        <th>
          Delay
        </th>
        <th>
          Alert?
        </th>
        <th>
          Deserves Alert?
        </th>
        <th>
          Alert Delay
        </th>
        <th>
          Alert Text
        </th>
        <th>
          Predicted Delay
        </th>
        <th>
          Delay Accuracy
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td>
          inbound
        </td>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          No
        </td>
        <td>
          No
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td>
          outbound
        </td>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          Yes
        </td>
        <td>
          Yes
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td>
          inbound
        </td>
        <td />
        <td>
          
        </td>
        <td>
          12:00 AM
        </td>
        <td>
          -15 min
        </td>
        <td>
          No
        </td>
        <td>
          No
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td />
        <td />
        <td>
          12:00 AM
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          No
        </td>
        <td>
          No
        </td>
        <td>
          5 min
        </td>
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td>
          04/26/2017
        </td>
        <td>
          12:00 AM
        </td>
        <td>
          Wednesday
        </td>
        <td />
        <td />
        <td />
        <td />
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          No
        </td>
        <td>
          No
        </td>
        <td>
          
        </td>
        <td />
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`AlertEventList should render data 3`] = `
<div
  className="table-responsive"
>
  <table
    className="table-sm mt-5 table table-bordered table-striped"
  >
    <thead
      className="thead-default"
    >
      <tr>
        <th>
          Trip ID
        </th>
        <th>
          Date
        </th>
        <th>
          Time
        </th>
        <th>
          Day
        </th>
        <th>
          Route
        </th>
        <th>
          Stop
        </th>
        <th>
          Direction
        </th>
        <th>
          Short Name
        </th>
        <th>
          Scheduled Dep.
        </th>
        <th>
          Actual Dep.
        </th>
        <th>
          Delay
        </th>
        <th>
          Alert?
        </th>
        <th>
          Deserves Alert?
        </th>
        <th>
          Alert Delay
        </th>
        <th>
          Alert Text
        </th>
        <th>
          Predicted Delay
        </th>
        <th>
          Delay Accuracy
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <td
          className="py-5"
          colSpan="17"
        >
          No Data Loaded
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
